corehq/apps/reports/views.py

```diff
@@ -312,6 +312,28 @@ def _is_valid(rn):
                 ret.append(scheduled_report)
         return sorted(ret, key=self._report_sort_key())
 
+    @property
+    def shared_saved_reports(self):
+        user = self.request.couch_user
+        config_reports = []
+
+        if user.is_domain_admin(self.domain):
+            # Admin user should see ALL shared saved reports
+            config_reports = ReportConfig.shared_on_domain(self.domain)
+        else:
+            # The non-admin user should ONLY see his/her saved reports (ie ReportConfigs) which have been used
+            # in a ReportNotification (not other users' ReportConfigs).
+            [config_reports.extend(r.configs) for r in self.scheduled_reports]
+
+        good_configs = []
+        for config in config_reports:
+            if config.is_configurable_report and not config.configurable_report:
```

Comment:
@proteusvacuum As far as I know there may exist some cases when a report is not available (see the comment in [this](https://github.com/dimagi/commcare-hq/blob/f69642fea47458bff467b73e69971cc2f00d955a/corehq/apps/saved_reports/models.py#L322) method; the method), but I mainly included it because of [this](https://github.com/dimagi/commcare-hq/blob/7b0ea8900b99b1be0bd0152b4321a170a5a1b282/corehq/apps/reports/views.py#L249) method that seems to be checking for exactly this case.